### ISSUE TITLE
Expect node to have attributes

### ISSUE THREAD

**Issue by Munter** [unexpectedjs/unexpected-jsdom#3]
That would be a nice convenience. I think an api similar to ‘expect(object, 'to have properties', { foo: 'bar' }‘ would make sense.

What do you think?

**Comment by papandreou** [comment]
Yeah, that would be really useful. I would like to have a separate library to do DOM query-based assertions, but I want it to work both in browsers and with jsdom, so perhaps it should be a separate module. Something similar to http://chaijs.com/plugins/chai-jquery

It has been on my (and @sunesimonsen's) table for some time, but for some reason none of us have actually needed it that much ourselves, so it hasn't yet materialized. I don't think it would be a huge undertaking, though.

**Comment by Munter** [comment]
I've started being quite strict in my unit tests at work. I'm now doing things like setting my react components in a state and verifying that the template is rendered as expected, which includes dom node attributes. However, currently I do it in a way that is not context aware, meaning possible error message will be something like "Expected `false` to be `true`", which is exactly the reason what I switched to unexpected to avoid :)

I haven't written an unexpected plugin myself yet, so I can take a stab at it with maybe a bit of guidance
